packages/proxy/src/providers/anthropic.test.ts

```diff
@@ -517,3 +517,60 @@ it("should handle file content parts with image data", async () => {
   expect(response!.choices[0].message.content).toBeTruthy();
   expect(typeof response!.choices[0].message.content).toBe("string");
 });
+
+it("should use model's max_output_tokens as default when max_tokens not specified", async () => {
+  // This test verifies BRA-3646: when max_tokens is not set, the proxy should
+  // use the model's max_output_tokens from the model spec instead of the
+  // hardcoded 4096 default. claude-sonnet-4-5 has max_output_tokens: 64000.
+  const { fetch, requests } = createCapturingFetch({ captureOnly: true });
+
+  await callProxyV1<OpenAIChatCompletionCreateParams, OpenAIChatCompletion>({
+    body: {
+      model: "claude-sonnet-4-5",
+      messages: [{ role: "user", content: "Hello" }],
+      stream: false,
+      // Intentionally NOT setting max_tokens - this is the key part of the test.
+      // Previously this would default to 4096.
+      // Now it should use the model's max_output_tokens (64000).
+    },
+    fetch,
+  });
+
+  expect(requests).toHaveLength(1);
+  // Verify the proxy sent max_tokens: 64000 (from model spec) instead of 4096
+  expect(requests[0].body).toMatchObject({ max_tokens: 64000 });
+});
+
+it("should return error when non-3.7 model receives max_tokens exceeding its limit", async () => {
+  const { statusCode } = await callProxyV1<
+    OpenAIChatCompletionCreateParams,
+    OpenAIChatCompletion
+  >({
+    body: {
+      model: "claude-sonnet-4-5",
+      messages: [{ role: "user", content: "Hello" }],
+      stream: false,
+      max_tokens: 128000,
+    },
+  });
+
+  expect(statusCode).toBeGreaterThanOrEqual(400);
+});
+
+it("should succeed with claude-3-7-sonnet when max_tokens unset (uses 128000 default + beta header)", async () => {
+  const { json, statusCode } = await callProxyV1<
+    OpenAIChatCompletionCreateParams,
+    OpenAIChatCompletion
+  >({
+    body: {
+      model: "claude-3-7-sonnet-latest",
+      messages: [{ role: "user", content: "Say hi" }],
+      stream: false,
```

Comment:
i would have used the custom fetch here and checked the request body and headers.